ggshield/cmd/utils/common_options.py

```diff
@@ -36,6 +38,15 @@ def get_config_from_context(ctx: click.Context) -> UserConfig:
     return cast(UserConfig, ctx.obj["config"].user_config)
 
 
+def check_directory_in_ignored_path(ctx: click.Context, directory: Path) -> None:
+    """Check if the directory is in the ignored paths in the config."""
+    if is_filepath_excluded(next(directory.rglob("*")), ctx.obj["exclusion_regexes"]):
+        display_warning(
+            "Directory to scan is in the ignored paths in the config, ignoring config and proceeding with scan."  # noqa: E501
+        )
+        ctx.obj["exclusion_regexes"] = set()
```

Comment:
I'm not sure completely ignoring the config is the expected behaviour, we should still be able to ignore some files even if we ask to scan a directory named `tests`. In my opinion we should just check the `exclusion_regexes` relatively to the scanned directory. I think this should be done in `ggshield/utils/files.py:get_filepaths`. 

As this is a common function to all verticals, we should check that this does not break anything.